diff --git a/packages/typescript/src/api/node/protocol.generated.ts b/packages/typescript/src/api/node/protocol.generated.ts
index 648eabeb..16b48dc7 100644
--- a/packages/typescript/src/api/node/protocol.generated.ts
+++ b/packages/typescript/src/api/node/protocol.generated.ts
@@ -6,7 +6,7 @@ export const childProperties: Readonly<Partial<Record<SyntaxKind, readonly (stri
     [SyntaxKind.QualifiedName]: ["left", "right"],
     [SyntaxKind.ComputedPropertyName]: ["expression"],
     [SyntaxKind.Decorator]: ["expression"],
-    [SyntaxKind.IfStatement]: ["expression", "thenStatement", "elseStatement"],
+    [SyntaxKind.IfStatement]: ["initializer", "expression", "thenStatement", "elseStatement"],
     [SyntaxKind.DoStatement]: ["statement", "expression"],
     [SyntaxKind.WhileStatement]: ["expression", "statement"],
     [SyntaxKind.ForStatement]: ["initializer", "condition", "incrementor", "statement"],
diff --git a/packages/typescript/src/ast/ast.generated.ts b/packages/typescript/src/ast/ast.generated.ts
index 76ddfe36..e8b13b29 100644
--- a/packages/typescript/src/ast/ast.generated.ts
+++ b/packages/typescript/src/ast/ast.generated.ts
@@ -523,6 +523,7 @@ export interface EmptyStatement extends StatementBase {
 }
 export interface IfStatement extends StatementBase {
     readonly kind: SyntaxKind.IfStatement;
+    readonly initializer?: ForInitializer;
     readonly expression: Expression;
     readonly thenStatement: Statement;
     readonly elseStatement?: Statement;
diff --git a/packages/typescript/src/ast/factory.generated.ts b/packages/typescript/src/ast/factory.generated.ts
index 787aa591..f4ddb523 100644
--- a/packages/typescript/src/ast/factory.generated.ts
+++ b/packages/typescript/src/ast/factory.generated.ts
@@ -761,7 +761,7 @@ function cloneNodeData(node: Node): any {
         case SyntaxKind.Decorator:
             return { expression: n.expression };
         case SyntaxKind.IfStatement:
-            return { expression: n.expression, thenStatement: n.thenStatement, elseStatement: n.elseStatement };
+            return { initializer: n.initializer, expression: n.expression, thenStatement: n.thenStatement, elseStatement: n.elseStatement };
         case SyntaxKind.DoStatement:
             return { statement: n.statement, expression: n.expression };
         case SyntaxKind.WhileStatement:
@@ -1132,6 +1132,7 @@ const forEachChildTable: Record<number, ForEachChildFunction> = {
     [SyntaxKind.ComputedPropertyName]: (data, cbNode, cbNodes) => visitNode(cbNode, data.expression),
     [SyntaxKind.Decorator]: (data, cbNode, cbNodes) => visitNode(cbNode, data.expression),
     [SyntaxKind.IfStatement]: (data, cbNode, cbNodes) =>
+        visitNode(cbNode, data.initializer) ||
         visitNode(cbNode, data.expression) ||
         visitNode(cbNode, data.thenStatement) ||
         visitNode(cbNode, data.elseStatement),
@@ -1717,8 +1718,9 @@ export function createEmptyStatement(): EmptyStatement {
     return new NodeObject(SyntaxKind.EmptyStatement, undefined) as unknown as EmptyStatement;
 }
 
-export function createIfStatement(expression: Expression, thenStatement: Statement, elseStatement?: Statement): IfStatement {
+export function createIfStatement(initializer: ForInitializer | undefined, expression: Expression, thenStatement: Statement, elseStatement?: Statement): IfStatement {
     return new NodeObject(SyntaxKind.IfStatement, {
+        initializer,
         expression,
         thenStatement,
         elseStatement,
@@ -3138,8 +3140,8 @@ export function updateDecorator(node: Decorator, expression: LeftHandSideExpress
     return node.expression !== expression ? createDecorator(expression) : node;
 }
 
-export function updateIfStatement(node: IfStatement, expression: Expression, thenStatement: Statement, elseStatement?: Statement): IfStatement {
-    return node.expression !== expression || node.thenStatement !== thenStatement || node.elseStatement !== elseStatement ? createIfStatement(expression, thenStatement, elseStatement) : node;
+export function updateIfStatement(node: IfStatement, initializer: ForInitializer | undefined, expression: Expression, thenStatement: Statement, elseStatement?: Statement): IfStatement {
+    return node.initializer !== initializer || node.expression !== expression || node.thenStatement !== thenStatement || node.elseStatement !== elseStatement ? createIfStatement(initializer, expression, thenStatement, elseStatement) : node;
 }
 
 export function updateDoStatement(node: DoStatement, statement: Statement, expression: Expression): DoStatement {
diff --git a/packages/typescript/src/ast/visitor.generated.ts b/packages/typescript/src/ast/visitor.generated.ts
index 00625a29..d1144224 100644
--- a/packages/typescript/src/ast/visitor.generated.ts
+++ b/packages/typescript/src/ast/visitor.generated.ts
@@ -527,10 +527,11 @@ const visitEachChildTable: Record<number, VisitEachChildFunction> = {
         return updateDecorator(node, _expression);
     },
     [SyntaxKind.IfStatement]: (node: IfStatement, visitor: Visitor): IfStatement => {
+        const _initializer = visitNode(node.initializer, visitor, isForInitializer);
         const _expression = visitNode(node.expression, visitor, isExpression);
         const _thenStatement = visitNode(node.thenStatement, visitor, isStatement);
         const _elseStatement = visitNode(node.elseStatement, visitor, isStatement);
-        return updateIfStatement(node, _expression, _thenStatement, _elseStatement);
+        return updateIfStatement(node, _initializer, _expression, _thenStatement, _elseStatement);
     },
     [SyntaxKind.DoStatement]: (node: DoStatement, visitor: Visitor): DoStatement => {
         const _statement = visitNode(node.statement, visitor, isStatement);
diff --git a/tsc/internal/api/encoder/decoder_generated.go b/tsc/internal/api/encoder/decoder_generated.go
index 6532a5a3..5f5b4255 100644
--- a/tsc/internal/api/encoder/decoder_generated.go
+++ b/tsc/internal/api/encoder/decoder_generated.go
@@ -225,10 +225,11 @@ func (d *astDecoder) createChildrenNode(kind ast.Kind, data uint32, childIndices
 		return d.factory.NewEmptyStatement(), nil
 	case ast.KindIfStatement:
 		it := newChildIter(childIndices)
-		expression := d.nodeAt(it.nextIf(mask, 0))
-		thenStatement := d.nodeAt(it.nextIf(mask, 1))
-		elseStatement := d.nodeAt(it.nextIf(mask, 2))
-		return d.factory.NewIfStatement(expression, thenStatement, elseStatement), nil
+		initializer := d.nodeAt(it.nextIf(mask, 0))
+		expression := d.nodeAt(it.nextIf(mask, 1))
+		thenStatement := d.nodeAt(it.nextIf(mask, 2))
+		elseStatement := d.nodeAt(it.nextIf(mask, 3))
+		return d.factory.NewIfStatement(initializer, expression, thenStatement, elseStatement), nil
 	case ast.KindDoStatement:
 		it := newChildIter(childIndices)
 		statement := d.nodeAt(it.nextIf(mask, 0))
diff --git a/tsc/internal/api/encoder/encoder_generated.go b/tsc/internal/api/encoder/encoder_generated.go
index d3e7599e..06f49f84 100644
--- a/tsc/internal/api/encoder/encoder_generated.go
+++ b/tsc/internal/api/encoder/encoder_generated.go
@@ -46,7 +46,7 @@ func getChildrenPropertyMask(node *ast.Node) uint8 {
 		return (boolToByte(n.Expression != nil) << 0)
 	case ast.KindIfStatement:
 		n := node.AsIfStatement()
-		return (boolToByte(n.Expression != nil) << 0) | (boolToByte(n.ThenStatement != nil) << 1) | (boolToByte(n.ElseStatement != nil) << 2)
+		return (boolToByte(n.Initializer != nil) << 0) | (boolToByte(n.Expression != nil) << 1) | (boolToByte(n.ThenStatement != nil) << 2) | (boolToByte(n.ElseStatement != nil) << 3)
 	case ast.KindDoStatement:
 		n := node.AsDoStatement()
 		return (boolToByte(n.Statement != nil) << 0) | (boolToByte(n.Expression != nil) << 1)
diff --git a/tsc/internal/ast/ast_generated.go b/tsc/internal/ast/ast_generated.go
index 87c1e59f..1ca2d4ce 100644
--- a/tsc/internal/ast/ast_generated.go
+++ b/tsc/internal/ast/ast_generated.go
@@ -819,41 +819,48 @@ func IsEmptyStatement(node *Node) bool {
 
 type IfStatement struct {
 	StatementBase
+	LocalsContainerBase
 	CompositeBase
+	Initializer   *ForInitializer // Optional
 	Expression    *Expression
 	ThenStatement *Statement
 	ElseStatement *Statement // Optional
 }
 
-func (f *NodeFactory) NewIfStatement(expression *Expression, thenStatement *Statement, elseStatement *Statement) *Node {
+func (f *NodeFactory) NewIfStatement(initializer *ForInitializer, expression *Expression, thenStatement *Statement, elseStatement *Statement) *Node {
 	data := f.ifStatementArena.New()
+	data.Initializer = initializer
 	data.Expression = expression
 	data.ThenStatement = thenStatement
 	data.ElseStatement = elseStatement
 	return f.newNode(KindIfStatement, data)
 }
 
-func (f *NodeFactory) UpdateIfStatement(node *IfStatement, expression *Expression, thenStatement *Statement, elseStatement *Statement) *Node {
-	if expression != node.Expression || thenStatement != node.ThenStatement || elseStatement != node.ElseStatement {
-		return updateNode(f.NewIfStatement(expression, thenStatement, elseStatement), node.AsNode(), f.hooks)
+func (f *NodeFactory) UpdateIfStatement(node *IfStatement, initializer *ForInitializer, expression *Expression, thenStatement *Statement, elseStatement *Statement) *Node {
+	if initializer != node.Initializer || expression != node.Expression || thenStatement != node.ThenStatement || elseStatement != node.ElseStatement {
+		return updateNode(f.NewIfStatement(initializer, expression, thenStatement, elseStatement), node.AsNode(), f.hooks)
 	}
 	return node.AsNode()
 }
 
 func (node *IfStatement) ForEachChild(v Visitor) bool {
-	return visit(v, node.Expression) || visit(v, node.ThenStatement) || visit(v, node.ElseStatement)
+	return visit(v, node.Initializer) ||
+		visit(v, node.Expression) ||
+		visit(v, node.ThenStatement) ||
+		visit(v, node.ElseStatement)
 }
 
 func (node *IfStatement) VisitEachChild(v *NodeVisitor) *Node {
-	return v.Factory.UpdateIfStatement(node, v.visitNode(node.Expression), v.visitEmbeddedStatement(node.ThenStatement), v.visitEmbeddedStatement(node.ElseStatement))
+	return v.Factory.UpdateIfStatement(node, v.visitNode(node.Initializer), v.visitNode(node.Expression), v.visitEmbeddedStatement(node.ThenStatement), v.visitEmbeddedStatement(node.ElseStatement))
 }
 
 func (node *IfStatement) Clone(f NodeFactoryCoercible) *Node {
-	return cloneNode(f.AsNodeFactory().NewIfStatement(node.Expression, node.ThenStatement, node.ElseStatement), node.AsNode(), f.AsNodeFactory().hooks)
+	return cloneNode(f.AsNodeFactory().NewIfStatement(node.Initializer, node.Expression, node.ThenStatement, node.ElseStatement), node.AsNode(), f.AsNodeFactory().hooks)
 }
 
 func (node *IfStatement) computeSubtreeFacts() SubtreeFacts {
-	return propagateSubtreeFacts(node.Expression) |
+	return propagateSubtreeFacts(node.Initializer) |
+		propagateSubtreeFacts(node.Expression) |
 		propagateSubtreeFacts(node.ThenStatement) |
 		propagateSubtreeFacts(node.ElseStatement)
 }
diff --git a/tsc/internal/ast/kind_stringer_generated.go b/tsc/internal/ast/kind_stringer_generated.go
index b80f1ba5..34999ff6 100644
--- a/tsc/internal/ast/kind_stringer_generated.go
+++ b/tsc/internal/ast/kind_stringer_generated.go
@@ -1,4 +1,4 @@
-// Code generated by "stringer -type=Kind -output=kind_stringer_generated.go"; DO NOT EDIT.
+// Code generated by "stringer -type=Kind -output=kind_stringer_generated.go ."; DO NOT EDIT.
 
 package ast
 
